tests/regression_tests/mg_temperature/build_2g.py

```diff
@@ -0,0 +1,291 @@
+                                                                    [t for t in
+                                                                     micro_cs],
+                                                                    micro_cs))
+    # Building a macro mg library
+    macro_xs = create_macro_dict(micro_cs)
+    mg_cross_sections_file_macro.add_xsdata(build_openmc_xs_lib('macro',
+                                                                groups,
+                                                                [t for t in
+                                                                 macro_xs],
+                                                                macro_xs))
+    # Exporting library to hdf5 files
+    mg_cross_sections_file_micro.export_to_hdf5('micro_2g.h5')
+    mg_cross_sections_file_macro.export_to_hdf5('macro_2g.h5')
+    # Returning the macro_xs dict is needed for analytical solution
+    return macro_xs
+
+
+def analytical_solution_2g_therm(xsmin, xsmax=None, wgt=1.0):
+    """ Calculate eigenvalue based on analytical solution for eq Lf = (1/k)Qf
+        in two group for infinity dilution media in assumption of group
+        boundary in thernmal spectra < 1.e+3 Ev
+        Parametres:
+        ----------
+        xsmin : dict
+            - macro cross-sections dictonary with minimum range temperature
+        xsmax : dict
+            - macro cross-sections dictonary with maximum range temperature
+              by default: None not used for standalone temperature
+        wgt : double
+            - weight for interpolation by default 1.0
+        Returns:
+        ---------
+        keff : np.double - analytical eigenvalue of critical eq matrix
+    """
+    if xsmax is None:
+        sa = xsmin['absorption']['macro']
+        ss12 = xsmin['scatter']['macro'][0][1][0]
+        nsf = xsmin['nu-fission']['macro']
+    else:
+        sa = xsmin['absorption']['macro'] * wgt + \
+             xsmax['absorption']['macro'] * (1 - wgt)
+        ss12 = xsmin['scatter']['macro'][0][1][0] * wgt + \
+               xsmax['scatter']['macro'][0][1][0] * (1 - wgt)
+        nsf = xsmin['nu-fission']['macro'] * wgt + \
+              xsmax['nu-fission']['macro'] * (1 - wgt)
+    L = np.array([sa[0] + ss12, 0.0, -ss12, sa[1]]).reshape(2, 2)
+    Q = np.array([nsf[0], nsf[1], 0.0, 0.0]).reshape(2, 2)
+    arr = np.linalg.inv(L).dot(Q)
+    return np.linalg.eigvals(arr)[1]
+
+
+def build_inf_model(xsnames, xslibname, temperature, tempmethod='nearest'):
+    """ Building an infinite medium for openmc multi-group testing
+        Parametres:
+        ----------
+        xsnames : list of str()
+            - list with xs names
+        xslibname:
+            - name of hdf5 file with cross-section library
+        temperature : float
+            - value of a current temperature in K
+        tempmethod : str {'nearest', 'interpolstion'} by default 'nearest'

```

Comment:
No need to state the type (str), just leave the options in the curly braces. Also move the default to the next line with a description.